spec/support/shared_examples/container.rb

```diff
@@ -185,6 +185,26 @@
           expect(container[:item].call).to eq('item')
         end
       end
+
+      context 'with option lazy_singleton: true' do
+        it 'registers and resolves a proc' do
+          container.register(:item, proc { 'item' }, lazy_singleton: true)
+
+          expect(container.keys).to eq(['item'])
+          expect(container.key?(:item)).to be true
+          expect(container.resolve(:item)).to eq('item')
+          expect(container[:item]).to eq('item')
+        end
```

Comment:
This spec doesn't really check if it's the memoized item, as you create new string objects. It would be better to use a local var like `item = 'item'` and do `expect(container[:item]).to be(item)` (notice the `be` part, which checks object's identity rather than equality)